### DISCUSSION TITLE
[FR] Combine Checkboxes, Dropdown, Multi-Select, Radios into a single field

### DISCUSSION THREAD

**Discussion by leevigraham** [craftcms/cms#13778] [Ideas]
Symphony has a single [`Choice Field`](https://symfony.com/doc/current/reference/forms/types/choice.html) which depending on two properties changes the display of the options.

Those options are:

* [`expanded`](https://symfony.com/doc/current/reference/forms/types/choice.html)
* [`multiple`](https://symfony.com/doc/current/reference/forms/types/choice.html)

This allows devs to change the field presentation without loosing the underlying options / data.

Currently in CraftCMS changing the field type clears the table of options… I'm not sure about the data.

It would be great if:

1. The table of options / data is not cleared while changing the field type or 
2. The field is combined into a single 'Choice' field

There is one challenge when changing from a single choice to a multiple choice and that's how to handle an array of choices when moving back to a single field. My guess would be the first item in the array becomes the value. CraftCMS already has a warning for changing field types and loosing data some a warning could work.

// @iainhenderson

**Comment by brandonkelly** [comment]
Checkboxes and Multi-select are interchangable, and Dropdown and Radio Buttons are interchangeable. (Multi-selection vs. single-selection.)

I don’t think it makes sense to start storing Dropdown/Radio button data as a JSON-encoded array, in the off chance that the field gets converted to a multi-select field type down the road. So at best we would be reducing them to two field types instead of four.

**Reply by leevigraham** [reply]
Interchangeable in regards to stored data… but changing the field in the backend reloads the settings table and blasts the previous values.

https://github.com/craftcms/cms/assets/25124/84fd5c49-3376-4c38-af19-00d585baf2d5

**Reply by brandonkelly** [reply]
We sortof solved that with condition rules. In that case, if you are switching a condition rule type, any identically-named settings will be retained between the types, if the setting properties are defined by the same base class. We should probably do the same thing with field types in Craft 5.
